src/Components/Firestore.js

```diff
@@ -201,3 +202,38 @@ export async function SaveListReactionsToFirestore(docId, updatedRxns) {
     console.error("Error writing review data to Firestore", error);
   }
 }
+
+export async function SaveHandle(handle, userId) {
+  let documentRef = doc(db, "handles", userId);
+  try {
+    await setDoc(
+      documentRef,
+      { handle: handle, userId: userId },
+      { merge: true }
+    );
+  } catch (error) {
+    console.error("Error writing handle to Firestore", error);
+  }
+}
+
+export async function CheckForHandleDuplicates(
+  handle,
+  user,
+  newLocalUser,
+  setLocalUser
+) {
+  try {
+    const collectionRef = collection(db, "handles");
+    const q = query(collectionRef, where("handle", "==", handle));
```

Comment:
Ok, just realized I was wrong about the first part -- the user *shouldn't* be able to query the `/users` collection for other users with the handle since they shouldn't be able to read `/users` documents for anyone other than themselves, since these contain private data.  So the current way you have it does work around that issue.  But if we do have intention of preventing those races at some point, we probably should still key `/handles` by normalized handle.  You can do that with or without the transaction in this PR, whichever you prefer.

Reply:
Awesome feedback - I like the idea of "future-proofing" this kind of stuff right now, so I included the transactions.